### PR TITLE
[HGCAL trigger] Additional cluster shape variables

### PR DESCRIPTION
**Pull request by jbsauvan**
#### PR description:
- Add several cluster shape variables meant to be implementable in firmware

Associated internal PRs and reviews:
- hgc-tpg#2
- hgc-tpg#47

Related presentation in HGCAL TPG meetings:
- [Study of alternative cluster shape variables](https://indico.cern.ch/event/1060474/#sc-7-2-study-of-alternative-cl)
- [Cluster shape variables quantization](https://indico.cern.ch/event/1122266/#3-updates-on-cluster-shape-var)
- [Updated studies on quantized shape variables](https://indico.cern.ch/event/1216136/#4-updated-studies-on-quantized)

#### PR validation:
Tested D86, D92, D9 workflows
```
runTheMatrix.py -w upgrade -l 20034.0 --maxSteps=2 & # D86
runTheMatrix.py -w upgrade -l 22434.0 --maxSteps=2 & # D92
runTheMatrix.py -w upgrade -l 23234.0 --maxSteps=2 & # D94
```

### CODE REVIEW

File: DataFormats/L1THGCal/interface/HGCalClusterT.h [hunk 1]
```diff
@@ -150,6 +150,10 @@ namespace l1t {
     float sigmaPhiPhiTot() const { return sigmaPhiPhiTot_; }
     float sigmaZZ() const { return sigmaZZ_; }
     float sigmaRRTot() const { return sigmaRRTot_; }
+    float varRR() const { return varRR_; }
+    float varZZ() const { return varZZ_; }
+    float varEtaEta() const { return varEtaEta_; }
+    float varPhiPhi() const { return varPhiPhi_; }
```
Comment (aloeliger) [inline DataFormats/L1THGCal/interface/HGCalClusterT.h:156]: I guess it isn't deal breaking for me, but it seems the difference between _getting_ variables and _setting_ variables in this code is whether or not an argument is provided to this overloaded function. Diametrically opposed functionality of an overloaded function is probably a bit dangerous. A bit more traditional would be to label these methods with `get` and `set` as appropriate.

Comment (jbsauvan) [reply DataFormats/L1THGCal/interface/HGCalClusterT.h:156]: Thanks @aloeliger 
I have prepended all setters in this class with ˋsetˋ to differentiate them from getters.

**Comment by aloeliger** [comment]
Please test

**Comment by aloeliger** [comment]
please test

**Comment by aloeliger** [comment]
+l1

**Comment by smuzaffar** [comment]
Milestone for this pull request has been moved to CMSSW_14_0_X.Please open a backport if it should also go in to CMSSW_13_3_X.

**Comment by srimanob** [comment]
@cmsbuild please test

Re-trigger the test after 2 weeks.

**Comment by jbsauvan** [comment]
Hello, 
Are there any additional comments on this PR?
Thanks a lot

**Comment by srimanob** [comment]
+Upgrade

**Comment by antoniovilela** [comment]
+1

**Pull request merged**